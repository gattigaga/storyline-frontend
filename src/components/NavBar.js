import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "react-router-dom";

const Container = styled.div`
  width: 100%;
  height: 48px;
  background: white;
  border-bottom: 1px solid #ddd;
  z-index: 999;
`;

const Wrapper = styled.div`
  width: 85%;
  display: flex;
  align-items: center;
  margin: auto;
`;

const Title = styled.h1`
  font-family: Roboto Slab;
  font-weight: bold;
  font-size: 24px;
  margin: 0px;
  color: #333;
`;

const List = styled.ul``;

const ListItem = styled.li`
  display: inline;
  margin-right: 24px;
`;

const StyledLink = styled(Link)`
  color: #333;
  font-family: Roboto Slab;
  font-size: 14px;
  text-decoration: none;

  &:hover {
    color: #e74c3c;
  }
`;

const NavBar = ({ items }) => (
  <Container>
    <Wrapper>
      <Title>STORYLINE</Title>
      <List>
        {items.map((item, index) => (
          <ListItem key={index}>
            <StyledLink to={item.url}>{item.label}</StyledLink>
          </ListItem>
        ))}
      </List>
    </Wrapper>
  </Container>
);

NavBar.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      url: PropTypes.string
    })
  )
};

export default NavBar;
