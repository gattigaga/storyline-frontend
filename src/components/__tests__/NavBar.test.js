import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import NavBar from "../NavBar";

describe("NavBar", () => {
  const setup = propOverrides => {
    const props = {
      items: [
        {
          label: "Home",
          url: "/"
        },
        {
          label: "Technology",
          url: "/topic/technology"
        },
        {
          label: "Science",
          url: "/topic/science"
        }
      ],
      ...propOverrides
    };

    const wrapper = shallow(<NavBar {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
