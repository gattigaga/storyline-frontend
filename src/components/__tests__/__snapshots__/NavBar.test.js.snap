// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should renders default 1`] = `
<styled.div>
  <styled.div>
    <styled.h1>
      STORYLINE
    </styled.h1>
    <styled.ul>
      <styled.li
        key="0"
      >
        <Styled(Link)
          to="/"
        >
          Home
        </Styled(Link)>
      </styled.li>
      <styled.li
        key="1"
      >
        <Styled(Link)
          to="/topic/technology"
        >
          Technology
        </Styled(Link)>
      </styled.li>
      <styled.li
        key="2"
      >
        <Styled(Link)
          to="/topic/science"
        >
          Science
        </Styled(Link)>
      </styled.li>
    </styled.ul>
  </styled.div>
</styled.div>
`;
